### DISCUSSION TITLE
How to clear/clean `.pnpm-store` cache

### DISCUSSION THREAD

**Discussion by ttodua** [pnpm/pnpm#4413] [Q&A]
In [docs,](https://pnpm.io/motivation) i couldnt find a way to force clean the cache for pnpm (either global `pnpm-store` or the project-scope `node_modules`). Any hints?

Note, `pnpm store prune` does not seem what I ask, because after running it, if you enter the directory (returned by `pnpm store path`) the foldes/files are still there.

**Comment by zkochan** [comment]
```
pnpm store prune
```

https://pnpm.io/cli/store#prune

**Reply by ttodua** [reply]
thanks, but it mentioned:
> Removes unreferenced packages from the store.

which is not what I asked.

**Reply by zkochan** [reply]
That is the only way to clean it. How would you expect to clean it? If you just remove the store, all the files will still remain in the node_modules of the project. Just removing the store will not clean up any space on the disk.

**Reply by ttodua** [reply]
I know, but for some unknown reasons we had many issues on continuous deployment, where (even though there were present package.json and also  pnpm-package-lock too) it was not installing some modules during `pnpm i`.  Only deleting `.pnpm-store` helped in all those cases. So, if you can, add an option to just delete that folder on demand.

**Reply by zkochan** [reply]
You can get the location of the store via `pnpm store path` and then just run `rm -rf`

**Reply by bryanprimus** [reply]
> I know, but for some unknown reasons we had many issues on continuous deployment, where (even though there were present package.json and also pnpm-package-lock too) it was not installing some modules during `pnpm i`

how did you resolve this? this happens to me as well. some peerDependencies not installed where I  already set `auto-install-peers=true`

**Reply by 121Unicorns** [reply]
Deleting all node_modules folder and running `pnpm store prune` worked for me. Thanks.

**Reply by cakadi190** [reply]
Thank you, it's work properly!

**Reply by sefatanam** [reply]
works for me as well

**Comment by deadcoder0904** [comment]
why not add how much disk size it cleared as well?

i always like to know how much it cleared. currenty, it looks like:

```bash
Removed all cached metadata files
Removed 9548 files
Removed 591 packages
```

but yeah i love when `brew` shows how much space cleared up in numbers.

**Reply by sheecegardezi** [reply]
you can go to the sotre directory and check how much space is being used:

```
cd ~/.local/share/pnpm/store
du -h
```

**Reply by deadcoder0904** [reply]
@sheecegardezi i know that but it should show it directly bcz why not?

**Comment by ttodua** [comment]
both location and command are different across OS-es,  that's why it's s simply necessary there was native command, like `pnpm --clear-cache` or whatever.

**Comment by its-dibo** [comment]
both npm and yarn offers a cmd to clear the cache

```
npm cache clean -f

yarn cache clean
``

**Reply by s-h-a-d-o-w** [reply]
I'm not sure that everybody realizes that unlike npm and yarn, pnpm uses symbolic links and that they'd end up with broken `node_modules` everywhere when simply deleting the store contents.

Which is probably why it's implemented the way it is at the moment - to ensure that people aren't still relying on dependencies from the store.

**Comment by luoye00** [comment]
This works for me

**Comment by avisek** [comment]
```bash
rm -rf "$(pnpm store path)"
```
